### ISSUE TITLE
How to run invoke-azurermvmruncommand

### ISSUE THREAD

**Issue by Rahulnigam28** [Azure/azure-sdk-for-python#3358]
Hi,

I am looking for python version of invoke-azurermvmruncommand.

I saw that currently in 2018-06-01, it supports only “get” and “list” methods for virtual machine run commands operation class.

It will be very helpful of you if you would let me know how to run invokeazurermvmrun command using python by invoking a script?

Thanks

**Comment by lmazuel** [comment]
Hi @Rahulnigam28 !

I think this StackOverflow answer I made recently answers your question
https://stackoverflow.com/a/51485744/4074838

Let me know otherwise!
Thanks!

**Comment by Rahulnigam28** [comment]
Thank you for your response. Can we also invoke a shell script located locally and pass arguments to that script? Or are we only allowed to run commands?

**Comment by lmazuel** [comment]
Please clarify "locally", do you mean locally on the VM, or locally on your machine?
Roughly, this connects to the VM, start a shell, and execute the command as-is. So anything you would be able to do in a SSH session would work. If you have a script on your local machine, you need to upload it on the VM (if it's always the same script you want on all VMs, this can be done as post-process in your deployment template, so you don't have to upload it later).
Does it help or did I miss something?

**Comment by Rahulnigam28** [comment]
I was able to that from azure automation runbook and  from a jumpbox server. I downloaded the script to my local VM from where I was running that invoke-azurerunvmrun command. I just give the path to  where that script gets downloaded on my local VM and then run invoke-azurermvmruncommand by passing script path parameters and arguments  without needing to download that script on VM first and then run it.
